### PR TITLE
[ADD] add new module: pos_edit_order

### PR DESCRIPTION
**Pull request by GabbasovDinar**

### CODE REVIEW

**Comment by legalsylvain** [comment]
Hi. Coule you add a description of the module ? Feature, etc... 
Thanks.

**Comment by francesco-ooops** [comment]
@GabbasovDinar please set PR as draft for now

**Review by francesco-ooops — APPROVED** [review]
Functional review ok

From description:

This module adds a button "Edit order lines" in main POS interface to allow an easier UX for cashier when editing product price, qty and discount. 

On button click, a popup with order lines allows user to input a clear value for each field, instead of having to use the default process of selecting which field to edit, which is not very intuitive and error-prone. 

@legalsylvain could you take a look? Thanks!

File: pos_edit_order_line/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+This module adds a button "Edit order lines" in main POS interface to allow an easier UX for cashier when editing product price, qty and discount.
+
+On button click, a popup with order lines allows user to input a clear value for each field, instead of having to use the default process of selecting which field to edit, which is not very intuitive and error-prone.
```
Comment (legalsylvain) [inline pos_edit_order_line/readme/DESCRIPTION.rst:3]: Nice to have : could you have a print screen ? and add reference to the print screen in this file ?

**Review by SicurSam — APPROVED** [review]
Functional review ok

File: pos_edit_order_line/models/pos_config.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class PosConfig(models.Model):
+    _inherit = "pos.config"
+
+    allow_edit_order_line = fields.Boolean(
+        default=False,
```
Comment (legalsylvain) [inline pos_edit_order_line/models/pos_config.py]: Well. When we install a module we expect we want the feature. So set default=True is better IMO.

File: pos_edit_order_line/readme/ROADMAP.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+TODO:
```
Comment (legalsylvain) [inline pos_edit_order_line/readme/ROADMAP.rst]: To remove, if there is nothing to write

File: pos_edit_order_line/readme/USAGE.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+TODO:
```
Comment (legalsylvain) [inline pos_edit_order_line/readme/USAGE.rst]: Same here.

**Comment by quentinDupont** [comment]
Hello and thanks for this module.
Could it be possible to had @francesco-ooops description in code with screenshot/gif of what this module does.
Thank you !

File: pos_edit_order_line/static/src/xml/EditOrderPopup.xml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                <main class="full-content">
+                    <div class="order-line-window">
+                        <section class="subwindow list">
+                            <div class="subwindow-container">
+                                <div class="subwindow-container-fix scrollable-y">
+                                    <table class="order-line-list">
+                                        <thead>
+                                            <tr>
+                                                <th>Product</th>
+                                                <th>Quantity</th>
```
Comment (legalsylvain) [inline pos_edit_order_line/static/src/xml/EditOrderPopup.xml:19]: Could you display the UoM in another colum ?

**Review by legalsylvain — COMMENTED** [review]
Hi. Thanks a lot for sharing this great module !

Functional review : works like a charm.

Some minor / nitpicking remarks inline.

Regards.

File: pos_edit_order_line/readme/CONFIGURE.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+Select PoS > configuration > enable flag "Allow Edit Order Line"
```
Comment (francesco-ooops) [inline pos_edit_order_line/readme/CONFIGURE.rst:1]: To improve usability when there are more than 3 order lines, we suggest activating the "Large scrollbars" POS setting.

**Comment by francesco-ooops** [comment]
Hi @legalsylvain , could you check if all your remarks have been addressed?

**Comment by legalsylvain** [comment]
Hi. All remarks OK. Juste tested on runboat. Works like a charm.

Non blocking UX remarks : what about to set gray the édit Burton when order is empty ? 

Otherwise lgtm.
Thanks !

**Review by legalsylvain — APPROVED** [review]

**Review by flotho — APPROVED** [review]
I confirm, like a charm on runboat

**Comment by francesco-ooops** [comment]
Hi @legalsylvain , in this PR a screenshot was requested to be added to description, but now seems broken: https://github.com/OCA/pos/tree/14.0/pos_edit_order_line

just a curiosity, do you know why this happens?

**Pull request merged**